ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/transactions/TransactionPoolTest.java

```diff
@@ -545,4 +583,44 @@ private void givenTransactionIsValid(final Transaction transaction) {
             eq(transaction), nullable(Account.class), anyBoolean()))
         .thenReturn(valid());
   }
+
+  @Test
+  public void shouldSendOnlyLocalTransactionToNewlyConnectedPeer() {
+    SyncState syncState = mock(SyncState.class);
+    when(syncState.isInSync(anyLong())).thenReturn(true);
+    EthProtocolManager ethProtocolManager = EthProtocolManagerTestUtil.create();
+    EthContext ethContext = ethProtocolManager.ethContext();
+    PeerTransactionTracker peerTransactionTracker = new PeerTransactionTracker();
+    TransactionPool transactionPool =
+        new TransactionPool(
+            transactions,
+            protocolSchedule,
+            protocolContext,
+            batchAddedListener,
+            syncState,
+            ethContext,
+            peerTransactionTracker);
+
+    final TransactionTestFixture builder = new TransactionTestFixture();
+    final Transaction transactionLocal = builder.nonce(1).createTransaction(KEY_PAIR1);
+    final Transaction transactionRemote = builder.nonce(2).createTransaction(KEY_PAIR1);
+    when(transactionValidator.validate(any(Transaction.class))).thenReturn(valid());
+    when(transactionValidator.validateForSender(
+            eq(transactionLocal), nullable(Account.class), eq(true)))
+        .thenReturn(valid());
+    when(transactionValidator.validateForSender(
+            eq(transactionRemote), nullable(Account.class), eq(true)))
+        .thenReturn(valid());
+    transactionPool.addLocalTransaction(transactionLocal);
+    transactionPool.addRemoteTransactions(Collections.singletonList(transactionRemote));
+
+    RespondingEthPeer peer = EthProtocolManagerTestUtil.createPeer(ethProtocolManager);
+
+    Set<Transaction> transactionsToSendToPeer =
+        peerTransactionTracker.claimTransactionsToSendToPeer(peer.getEthPeer());
+
+    assertThat(transactionsToSendToPeer.size()).isEqualTo(1);
```

Comment:
I think this can be:
```suggestion
    assertThat(transactionsToSendToPeer).containsExactly(transactionLocal);
```

Reply:
And then remove the line below that checks the hash.